src/ast/passes/fold_literals.cpp

```diff
@@ -440,6 +444,31 @@ std::optional<Expression> LiteralFolder::visit(Expression &expr)
   return std::nullopt;
 }
 
+std::optional<Expression> LiteralFolder::visit(Probe &probe)
+{
+  top_level_node_ = &probe;
+  return Visitor<LiteralFolder, std::optional<Expression>>::visit(probe);
+}
+
+std::optional<Expression> LiteralFolder::visit(Builtin &builtin)
+{
+  if (builtin.ident == "usermode") {
+    if (auto *probe = dynamic_cast<Probe *>(top_level_node_)) {
+      for (auto *ap : probe->attach_points) {
+        if (!ap->check_available(builtin.ident)) {
```

Comment:
For functions that aren't available for the probe it looks like we add an error and don't let the program continue. Any reason why we shouldn't do that here too?

Reply:
Actually, nm, I realized this was a dumb question.